src/game/game.controller.ts

```diff
@@ -0,0 +1,131 @@
+import { Request, Response } from 'express';
+import { instanceToPlain, plainToInstance } from 'class-transformer';
+import { validate } from 'class-validator';
+import { Controller } from '../common';
+import { GameService, AddGameDto, AddLevelDto } from '.';
+import { StatusCodes, ReasonPhrases } from 'http-status-codes';
+
+export class GameController extends Controller {
+    constructor() {
+        super('/game');
+
+        const router = this.getRouter();
+
+        router.get('/', this.getAllGames);
+        router.post('/', this.addGame);
+        router.get('/:id', this.getGame);
+        router.delete('/:id', this.removeGame);
+        router.post('/:id/experience', this.addExperienceLevel);
+        router.delete('/:id/experience/:lvl_id', this.removeExperienceLevel);
+    }
+
+    async getAllGames(req: Request, res: Response) {
+        return res.send(await GameService.getAllGames());
+    }
+
+    async getGame(req: Request, res: Response) {
+        const id = req.params.id;
+
+        try {
+            const game = await GameService.getGame(id);
+
+            if (!game) {
+                return res.status(StatusCodes.NOT_FOUND).send('Game not found');
+            }
+            return res.json(instanceToPlain(game));
+        } catch (error) {
+            console.error(error);
+            return res
+                .status(StatusCodes.INTERNAL_SERVER_ERROR)
+                .send(ReasonPhrases.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    async addGame(req: Request, res: Response) {
+        const body = plainToInstance(AddGameDto, req.body as AddGameDto);
+        const errors = await validate(body);
+        if (errors.length) {
+            return res.status(StatusCodes.BAD_REQUEST).json(errors);
+        }
+
+        const added = await GameService.addGame(body);
+
+        return res.status(StatusCodes.CREATED).send(instanceToPlain(added));
+    }
+
+    async removeGame(req: Request, res: Response) {
+        const id = req.params.id;
+
+        try {
+            const removed = await GameService.removeGame(id);
+
+            if (!removed) {
+                return res.status(StatusCodes.NOT_FOUND).send('Game not found');
+            }
+            return res
+                .status(StatusCodes.NO_CONTENT)
+                .send('Game has been removed');
```

Comment:
NO_CONTENT 204 should be just OK 200
also `Game has been removed` is not essential. we have status code

change that to `return res.send();`